### PR TITLE
Fix ordering in transaction API for MySQL

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

Similar fix to #1516. Only keep the latest operations for each key and ignore the other operations. This ensures that the ordering of the operations is intact.

## Issue reference

Please reference the issue this PR will close: #1521

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* (NA) Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/mysql/mysql.go [hunk 1]
```diff
@@ -523,6 +535,34 @@ func (m *MySQL) Multi(request *state.TransactionalStateRequest) error {
 	return nil
 }
 
+// Returns the set requests.
+func (m *MySQL) getSets(req state.TransactionalStateOperation) (state.SetRequest, error) {
```
Comment (daixiang0) [inline state/mysql/mysql.go:539]: Rename as `getSetRequest`

File: state/mysql/mysql.go [hunk 2]
```diff
@@ -523,6 +535,34 @@ func (m *MySQL) Multi(request *state.TransactionalStateRequest) error {
...
+
+	if setReq.Key == "" {
+		return setReq, fmt.Errorf("missing key in upsert operation")
+	}
+
+	return setReq, nil
+}
+
+// Returns the delete requests.
+func (m *MySQL) getDeletes(req state.TransactionalStateOperation) (state.DeleteRequest, error) {
```
Comment (daixiang0) [inline state/mysql/mysql.go:553]: Rename as `getDeleteRequest`

File: state/mysql/mysql.go (hunk 1 above)
Comment (shubham1172) [reply state/mysql/mysql.go:539]: This is the same convention we have been following for other state stores. Should we keep it for the sake of uniformity, or do you think it's better to update? Same is valid for the `getDeletes` comment.

For example https://github.com/dapr/components-contrib/blob/99df8741ea84ffcb973441efa858a26e8a46dda2/state/sqlserver/sqlserver.go#L357

Comment (daixiang0) [reply state/mysql/mysql.go:537]: OK.

File: state/mysql/mysql.go [hunk 3]
```diff
@@ -341,7 +341,28 @@ func (m *MySQL) deleteValue(req *state.DeleteRequest) error {
...
+
+	tx, err := m.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	if len(req) > 0 {
+		for _, d := range req {
+			da := d // Fix for goSec G601: Implicit memory aliasing in for loop.
+			err = m.Delete(&da)
```
Comment (daixiang0) [inline state/mysql/mysql.go:354]: Not sure whether we can use &req[i] here.

Comment (shubham1172) [reply state/mysql/mysql.go:354]: You mean, instead of `da := d`? Both of them can fix it IMO but I used this pattern for consistency with our codebase.

Comment (shubham1172) [reply state/mysql/mysql.go:354]: /cc @daixiang0

**Review by artursouza — APPROVED** [review]

**Pull request merged**